### PR TITLE
Fix testAutoExpandIndicesDuringRollingUpgrade

### LINKED ISSUE #50426: RecoveryIT.testAutoExpandIndicesDuringRollingUpgrade failing
Build scan:

https://gradle-enterprise.elastic.co/s/ad7pjkaf7e2h4/

Test mute incoming.

### PR DESCRIPTION
**Pull request by ywelsch**
Fixes the muted test "testAutoExpandIndicesDuringRollingUpgrade". We can't wait in the test for the index to be green, as we have put a filter exclusion into place that prevents all shards from being allocated after a node rejoins. Instead we check whether the correct auto-expansion has taken place. We need to do this in an assertBusy, as older versions of ES are not auto-expanding indices while a node is joining, but only doing this as a follow-up step.

Closes #50426

### CODE REVIEW

File: qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java [hunk 1]
```diff
@@ -796,15 +795,15 @@ public void testAutoExpandIndicesDuringRollingUpgrade() throws Exception {
...
-            getIndexSettingsAsMap(indexName).get(IndexMetaData.SETTING_NUMBER_OF_REPLICAS).toString());
-        if (minimumNodeVersion.onOrAfter(Version.V_7_6_0)) {
-            assertEquals(nodes.size() - 2, numberOfReplicas);
-        } else {
-            assertEquals(nodes.size() - 1, numberOfReplicas);
-        }
+        assertBusy(() -> {
+            final int numberOfReplicas = Integer.parseInt(
+                getIndexSettingsAsMap(indexName).get(IndexMetaData.SETTING_NUMBER_OF_REPLICAS).toString());
+            if (minimumNodeVersion.onOrAfter(Version.V_7_6_0)) {
```
Comment (DaveCTurner) [inline qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java]: Do we need the `assertBusy()` if `minimumNodeVersion.onOrAfter(Version.V_7_6_0)` or can we `ensureGreen()` in that case?

**Review by DaveCTurner — COMMENTED** [review]
I left a question

File: qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java (hunk 1 above)
Comment (ywelsch) [reply qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java]: Thanks for the question. We actually don't need assertBusy at all (https://github.com/elastic/elasticsearch/pull/30423 was already introduced in 6.4, I misremembered this).

The `ensureGreen` is mostly for cosmetics. I've added it for the case where we are on 7.6.0+

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**